### PR TITLE
feat(api-client): Login with access token

### PR DESCRIPTION
**Pull request by bennycode**

### CODE REVIEW

File: packages/api-client/src/APIClient.test.browser.ts [hunk 1]
```diff
@@ -21,12 +21,13 @@
 
 import {APIClient} from './APIClient';
 import {ClientType} from './client';
+import {AccessTokenData} from './auth';
```
Comment (ffflorian) [inline packages/api-client/src/APIClient.test.browser.ts:24]:
```suggestion
import type {AccessTokenData} from './auth';
```

File: packages/api-client/src/auth/AccessTokenData.ts [hunk 2]
```diff
@@ -18,8 +18,10 @@
  */
 
 export interface AccessTokenData {
+  /** Token format is described in: https://wire-docs.wire.com/download/Wire+Security+Whitepaper.pdf */
   access_token: string;
+  /** Expiration in seconds. */
```
Comment (ffflorian) [inline packages/api-client/src/auth/AccessTokenData.ts:23]:
```suggestion
  /** Expiration in seconds */
```

File: packages/api-client/src/auth/parseAccessToken.test.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import {AccessTokenType, parseAccessToken, parseValue} from './parseAccessToken';
```
Comment (ffflorian) [inline packages/api-client/src/auth/parseAccessToken.test.ts:1]: Missing copyright header

File: packages/api-client/src/auth/parseAccessToken.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
+export enum AccessTokenType {
```
Comment (ffflorian) [inline packages/api-client/src/auth/parseAccessToken.ts:1]: Missing copyright header

File: packages/api-client/src/auth/parseAccessToken.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+export interface AccessTokenInfo {
+  clientId: string;
+  expirationDate: Date;
+  keyIndex: number;
+  tag: string;
+  token: string;
+  type: AccessTokenType;
+  /** User data only exists in fields from cookies (i.e. "zuid"). */
```
Comment (ffflorian) [inline packages/api-client/src/auth/parseAccessToken.ts:13]:
```suggestion
  /** User data only exists in fields from cookies (e.g. "zuid") */
```

File: packages/api-client/src/obfuscation/ObfuscationUtil.test.node.ts [hunk 6]
```diff
@@ -20,9 +20,10 @@
 import {Cookie as ToughCookie} from 'tough-cookie';
 
 import {ObfuscationUtil} from './ObfuscationUtil';
+import {AccessTokenData} from '../auth';
```
Comment (ffflorian) [inline packages/api-client/src/obfuscation/ObfuscationUtil.test.node.ts:23]:
```suggestion
import type {AccessTokenData} from '../auth';
```

**Review by ffflorian — CHANGES_REQUESTED** [review]

File: packages/api-client/src/auth/parseAccessToken.ts (hunk 4 above)
Comment (bennycode) [reply packages/api-client/src/auth/parseAccessToken.ts:1]: That's my Kryptonite. 🙈

File: packages/api-client/src/auth/parseAccessToken.test.ts (hunk 3 above)
Comment (bennycode) [reply packages/api-client/src/auth/parseAccessToken.test.ts:1]: Time to add: https://github.com/nikku/eslint-plugin-license-header 😎

**Pull request merged**